src/librustc_mir/borrow_check/nll/type_check/relate_tys.rs

```diff
@@ -85,13 +85,24 @@ pub(super) fn relate_type_and_user_type<'tcx>(
     // variance to get the right relationship.
     let v1 = ty::Contravariant.xform(v);
 
-    TypeRelating::new(
+    let mut type_relating = TypeRelating::new(
         infcx.tcx,
         NllTypeRelatingDelegate::new(infcx, borrowck_context, locations, category),
         v1,
         b_variables,
```

Comment:
**b_variables**
Can you explain to me how this magic happens ? b.variables is written as b_variables ?

Reply:
b is destructured just above this.

Reply:
Would you mind pointing me to the line please ?

Reply:
line 79